scripts/build-and-test.sh

```diff
@@ -19,10 +19,21 @@ function run_tests_with_gdb {
   for x in bin/**/*-tests; do scripts/run-with-gdb.sh "$x"; done
 }
 
+function run_tests {
+    for x in bin/**/*-tests; do "$x"; done
+}
+
+num_procs=1
+if [[ "$(uname)" == "Darwin" ]]; then
+    num_procs=$(sysctl -n hw.ncpu)
+elif [[ "$(expr substr $(uname -s) 1 5)" == "Linux" ]]; then
+    num_procs=$(lscpu -p | grep -v '^#' | sort -u -t, -k 2,4 | wc -l) # physical cores
```

Comment:
Why not threads? e.g. my quad core i7 has 8 threads and I typically use -j8 for scons etc. Have I been "doing it wrong? ?